src/frontend/sharedComponents/BottomSheetModal/index.tsx

```diff
@@ -22,16 +22,19 @@ export const useBottomSheetModal = ({
 }) => {
   const initiallyOpenedRef = React.useRef(false);
   const sheetRef = React.useRef<RNBottomSheetModal>(null);
+  const isOpen = React.useRef<boolean>(false);
```

Comment:
i'm thinking that it would makes sense for this to be a state variable. might require some fixing up in other parts of the PR but should be straightforward enough using effects

Reply:
This will cause unecessary re-renders as a useState Hook. Everytime the state of a useState hook is changed, all the associated components re-render. In this case we dont need it to re-render because open sheet and close sheet is already causing that re-render, we just want to simply encapsulate that state of being open or close.

Lets say we did have a state hook: `[isOpen, setIsOpen] = useState()`
if we called `openSheet` this will cause the bottom sheet to rerender (desired behaviour),
Then we would call setIsOpen(true). This will then cause the bottom sheet to re-render again (which is not necessary, as none of the UI is changing)

That is why I am using a useRef, because it does not cause any rerenders when its state changes

Reply:
based on how you want use the opened state in this pr, it feels like the re-render caused by the open state changing is desirable (refer to the other comment to understand why).

more tangentially, i'm remembering why i didn't include this in the hook – the BottomSheetModal was built with the assumption that it would always be it's own navigation screen, which made having the variable `isOpen` unnecessary (especially since it's easy to just implement from the consumer side, which is what you had before)

Reply:
either way, fine to keep the current implementation that you have if it works

Reply:
think my main point with this is that as a consumer, it'd potentially be nice to able to write effects that are based on the `isOpened` value i.e.

```ts
const { isOpened } = useBottomModalSheet()

React.useEffect(() => {
  // ...
}, [isOpened)

```

not entirely sure, but I don't think the above is possible if `isOpened` is represented as a ref value instead of state, but could be mistaken (would be easy to test but I don't have the energy at the moment 😅)

Reply:
Okay, I see what you are saying about that rerender. But I think for the reusability of the component it, in most other use cases, we dont want that rerender.

Yeah, that makes sense why it wasnt originally included in the hook. Its also interesting that it is not included out of the box. But probably for the same reason as you stated!